cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -278,8 +296,15 @@ visualization_msgs::MarkerArray MapBuilderBridge::GetConstraintList() {
       color_residual.a = 1.0;
       color_residual.r = 1.0;
     } else {
-      constraint_marker = &constraint_inter_marker;
-      residual_marker = &residual_inter_marker;
+      if (constraint.node_id.trajectory_id ==
+          constraint.submap_id.trajectory_id) {
+        constraint_marker = &constraint_inter_same_trajectory_marker;
```

Comment:
How about making these constraints bright yellow and the cross-trajectory ones bright orange? What do you think?